site-scrapers/TrinityEMS/index.js

```diff
@@ -0,0 +1,253 @@
+    // No appointments available if defined
+    if (noDatesAvailable != undefined) {
+        return [];
+    } else {
+        let days = [];
+        try {
+            days = await page.$$(".activeday");
+        } catch (error) {
+            console.log(`error trying to get day numbers: ${error}`);
+        }
+        return days;
+    }
+}
+
+/**
+ * Fetches the small snippet of HTML which would appear in a popup when an active day is clicked by a human.
+ *
+ * As yet, it's not know how the slot count will be presented. See parseHTMLforSlotCount().
+ * @param {*} page
+ * @param {String} dateStr -- yyyy-mm-dd, as in '2021-04-26'
+ * @returns response text of available times XHR query
+ */
+async function getSlotsForDate(page, dateStr) {
+    const url = [
+        "https://app.acuityscheduling.com/schedule.php?action=availableTimes",
+        "showSelect=0",
+        "fulldate=1",
+        "owner=21713854",
+        "type=19620839",
+        "calendar=5109380",
+        `date=${dateStr}`,
+        "ignoreAppointment=",
+    ].join("&");
+
+    return await page.evaluate(async (url) => {
+        const response = await fetch(url);
+        const text = await response.text();
+        return parseHTMLforSlotCount(text);
+    }, url);
+}
+
+/**
+ * Advances the page to the next month by clicking on the "next" button.
+ *
+ * @param {*} page
+ */
+async function advanceMonth(page) {
+    const nextMonthLinkClass = ".calendar-next";
+
+    try {
+        await Promise.all([
+            await page.click(nextMonthLinkClass),
+            // If the timeout option is not set, the timeout (30000 ms) will expire, and
+            // no results will be returned. Setting it to something shorter yields results.
+            // 5000 seems to be too much because the pages return nearly instantaneously.
+            // But will leave it at that just in case of ... what?
+            await page.waitForSelector(nextMonthLinkClass, {
+                timeout: 5000,
+            }),
+        ]);
+    } catch (error) {
+        console.log(`${site.name} :: trouble advancing button: ${error}`);
+    }
+}
+
+/**
+ * Accumulates a month's worth of availability into the results object.
+ *
+ * @param {availability Object} dailySlotsForMonth -- a month's collection of availability by date
+ */
+function accumulateAvailabilityForMonth(results, dailySlotsForMonth) {
+    if (
+        dailySlotsForMonth == null ||
+        Object.keys(dailySlotsForMonth).length == 0
+    ) {
+        return;
+    }
+
+    try {
+        let accumulatedAvailability = results.availability;
+        Object.assign(
+            results.availability,
+            accumulatedAvailability,
+            dailySlotsForMonth
+        );
+        results.hasAvailability = true;
+    } catch (error) {
+        console.log(`${site.name} :: error trying to merge results: ${error}`);
+    }
+}
+
+/**
+ * TODO: this is just a guess as to how the popup will present the number of appointments.
```

Comment:
if this is a guess, I suggest also posting to Slack or S3 to help us see what things look like when appointments are available. See util/s3.js and util/slack.js for methods.

Reply:
Removing all current code in the method. Adding s3.savePageContent, hoping that it catches the HTML which describes the availability.